devito/arch/compiler.py

```diff
@@ -237,6 +228,21 @@ def __new_with__(self, **kwargs):
     def name(self):
         return self.__class__.__name__
 
+    @property
+    def version(self):
+        if self.suffix is not None:
+            try:
+                version = Version(str(float(self.suffix)))
+            except (TypeError, ValueError):
+                version = Version(self.suffix)
+        else:
+            try:
+                # Knowing the version may still be useful to pick supported flags
+                version = sniff_compiler_version(self.CC)
+            except (FileNotFoundError, OSError):
```

Comment:
in case of FileNotFoundError a `sys.exit()` will be called inside `sniff_compiler_version` and the program will abort, so we might have to tweak that too

Reply:
or we just remove this try-except and wait for the next use case to break

Reply:
Changed to something bit simpler